src/decode_all.rs

```diff
@@ -55,7 +55,19 @@ mod tests {
 					);
 
 					encoded.extend(&[1, 2, 3, 4, 5, 6]);
-					assert_eq!(<$type>::decode_all(&encoded).unwrap_err().what(), DECODE_ALL_ERR_MSG);
+					assert_eq!(
+						<$type>::decode_all(&encoded).unwrap_err().what(),
+						{
+							#[cfg(feature = "chain-error")]
+							{
+								"Input buffer has still data left after decoding!\n"
```

Comment:
I think it would be possible to get rid of this newline if you print it before each new cause if `indent != 0` instead of after

Reply:
I think it is better to go to new line if the error is multiline thus I did https://github.com/paritytech/parity-scale-codec/pull/242/commits/e597d83b5c0a20a802f1fc7e8f9a656d2a313170